src/app/public/modules/text-highlight/text-highlight.directive.ts

```diff
@@ -147,4 +147,14 @@ export class SkyTextHighlightDirective
       this.observer.observe(this.el.nativeElement, config);
     }
   }
+
+  private static cleanRegex(regex: string) {
+    return regex
+      .replace(/\\/g, '\\\\')
+      .replace(/\(/g, '\\(')
```

Comment:
We should probably accommodate some more characters:
```
\ ^ $ * + ? . ( ) | { } [ ]
```

Here's an example function that might work?
https://stackoverflow.com/a/3561711/6178885